lib/elixir/lib/integer.ex

```diff
@@ -238,67 +238,51 @@ defmodule Integer do
   def parse("", base) when base in 2..36,
     do: :error
 
-  def parse(binary, base) when is_binary(binary) and base in 2..36 do
-    parse_in_base(binary, base)
+  def parse(<<bin::binary>>, base) when base in 2..36 do
+    parse_sign(bin, base)
```

Comment:
Would

```elixir
{sign, rest} = parse_sign(bin)
parse_digits(rest, base, sign)
```

work in the same way? We can keep it a bit more tidy by not calling `parse_digits/3` from `parse_sign/2`. But maybe we do bad stuff with the binary context this way 🤔